tests/test_encoder_decoder_text_summarization.py

```diff
@@ -0,0 +1,99 @@
+import json
+import re
+import subprocess
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+import pytest
+
+from .test_examples import TIME_PERF_FACTOR
+
+
+MODELS_TO_TEST = {
+    "bf16": [
+        ("facebook/bart-large-cnn", 6.643, 32.399),
+        # ("gpt2-xl", 126.6292071377241),
+        # # TODO: fix OPT 6.7B
+        # # ("facebook/opt-6.7b", 0.0),
+        # ("EleutherAI/gpt-j-6b", 37.14562499113717),
+        # # TODO: GPT-NeoX doesn't fit on 1 Gaudi1 card
+        # # ("EleutherAI/gpt-neox-20b", 0.0),
+        # ("meta-llama/Llama-2-7b-hf", 43.951804139391925),
+        # ("tiiuae/falcon-7b", 44.288602257903726),
+        # ("bigcode/starcoder", 15.955986010526113),
+        # ("Salesforce/codegen2-1B", 109.03016111561857),
+        # ("mosaicml/mpt-7b", 44.888696119070424),
+    ],
+    # "deepspeed": [
+    #     ("bigscience/bloomz-7b1", 27.34439410425298),
+    # ],
+}
+
+
+def _test_text_summarization(
+    model_name: str, baseline: float, baseline_acc: float, token: str, deepspeed: bool = False, world_size: int = 8
+):
+    command = ["python3"]
+    path_to_example_dir = Path(__file__).resolve().parent.parent / "examples"
+
+    if deepspeed:
+        command += [
+            f"{path_to_example_dir / 'gaudi_spawn.py'}",
+            "--use_deepspeed",
+            f"--world_size {world_size}",
+        ]
+
+    command += [
+        f"{path_to_example_dir / 'summarization' / 'run_summarization.py'}",
+        f"--model_name_or_path {model_name}",
+        "--do_predict",
+        "--predict_with_generate",
+        "--dataset_name cnn_dailymail",
+        "--dataset_config 3.0.0",
+        "--use_habana",
+        "--per_device_eval_batch_size 2",
+        "--gaudi_config_name Habana/bart",
+        "--num_beams 1",
+        "--generation_num_beams 1",
```

Comment:
Why do we need to set both num_beams and generation_num_beams? Make this configurable so that we can run test with Beam search also.